### PR TITLE
Use JUnit 5 for tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/SignalR/issues/2802

### CODE REVIEW

File: clients/java/signalr/build.gradle [hunk 1]
```diff
@@ -16,7 +16,9 @@ repositories {
 }
 
 dependencies {
-    testImplementation group: 'junit', name: 'junit', version: '4.12'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.3.1'
```
Comment (mikaelm12) [inline clients/java/signalr/build.gradle]: Get these approved with @Eilon

File: clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java [hunk 2]
```diff
@@ -47,14 +42,12 @@ public void hubConnectionClosesAfterCloseMessage() throws Exception {
...
     public void hubConnectionReceiveHandshakeResponseWithError() throws Exception {
-        exceptionRule.expect(HubException.class);
-        exceptionRule.expectMessage("Requested protocol 'messagepack' is not available.");
-
         MockTransport mockTransport = new MockTransport();
         HubConnection hubConnection = new HubConnection("http://example.com", mockTransport, true);
 
         hubConnection.start();
-        mockTransport.receiveMessage("{\"error\":\"Requested protocol 'messagepack' is not available.\"}" + RECORD_SEPARATOR);
+        Throwable exception = assertThrows(HubException.class, () -> mockTransport.receiveMessage("{\"error\":\"Requested protocol 'messagepack' is not available.\"}" + RECORD_SEPARATOR));
```
Comment (mikaelm12) [inline clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java]: AssertThrows 😄

File: clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java [hunk 3]
```diff
@@ -649,7 +641,7 @@ public void receiveHandshakeResponseAndMessage() throws Exception {
         HubConnection hubConnection = new HubConnection("http://example.com", mockTransport, true);
 
         hubConnection.on("inc", () ->{
-            assertEquals(0.0, value.get(), 0);
+            assertEquals(Double.valueOf(0.0), value.get());
```
Comment (mikaelm12) [inline clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java]: Can it be `Double.valueOf(0)`

File: clients/java/signalr/build.gradle (hunk 1 above)
Comment (Eilon) [reply clients/java/signalr/build.gradle]: Approved all 3. Thanx!

Comment (mikaelm12) [reply clients/java/signalr/build.gradle]: Awesome. Thank you!

**Pull request merged**